src/index.ts

```diff
@@ -387,4 +421,21 @@ export default class OtaClient {
     private async getJsonFiles(file?: string): Promise<string[]> {
         return (await this.listFiles()).filter(f => !file || file === f).filter(isJsonFile);
     }
+
+    private async getLanguages(): Promise<Language[]> {
+        if (this.languagesCache) {
+            return this.languagesCache;
+        }
+        const languages = (await this.httpClient.get<APIData>(this.apiURL)).data.map(l => l.data);
```

Comment:
Why do you think it will be slower? You will just store `Promise` that we be eventually `resolved`. And this is the crucial part. If second time when you will call this method and in cache promise is already resolved you then will not lose any performance since it's already loaded. And also it will handle those situations when method will be called in parallel (which likely is not an rare situation). Right now this code will produce a lot of unneeded requests. So the whole point of caching is then lost here.

Reply:
Like I said, if the languages are cached this method will be a no-op, meaning that no request will be made (as you can see by the first few lines where we return the cache if one is found. Storing this as a promise will be attempting to resolve it every time when it could already be stored.

Reply:
`stringCache` is a little bit different because it holds strings per file

Reply:
well actually not) the structure is a bit more complex but the actual `value` inside per file and language is again `Promise`

Reply:
Can you please provide someking of cookbook where it is described that it's not a good practice? Do not get me wrong, all other changes are looking good to me and in some of the places I found quite a lot of new stuff for myself. And I am very grateful to you for that. But here....I do have my strong arguments why it needs to be updated. It's not about that or that looks better but actually to cover a possible performance bottleneck. Hope you will find this reasonable :)

Reply:
I can't really find any articles on this online and I think that's because this is such an unusual style of programming. The functionality, in the end, is the same, but it's harder to read and understand having to await a property than it is to await a method IMO. Also, since this is a private property, it doesn't make any difference in the end since you'd only be awaiting it once pretty much. I'm just going with what is more natural in my eyes. If you really really want me to change this, I'll do so, but I don't think we should

Reply:
Either we can fix it in this PR or let’s merge your changes and then for the release I will fix it by myself

Reply:
I’ll do sone tests later today and fix it if it needs fixing, along with the other review

Reply:
And actually it literally about changing 2 line of code. Adding Promise to the type definition and removing await in the method. That’s all.

Reply:
And not only the problem that I am trying to explain to you will be fixed but also it will be consistent with other caching (for manifest and strings)

Reply:
Managed to solve the issue, pushing it in a bit